src/contract_client/utils.ts

```diff
@@ -6,12 +8,32 @@ export const DEFAULT_TIMEOUT = 5 * 60;
 /**
  * Keep calling a `fn` for `timeoutInSeconds` seconds, if `keepWaitingIf` is true.
  * Returns an array of all attempts to call the function.
+ * @param fn Function to call repeatedly
+ * @param keepWaitingIf Condition to check when deciding whether or not to call `fn` again
+ * @param timeoutInSeconds How long to wait between the first and second call
+ * @param exponentialFactor What to multiply `timeoutInSeconds` by, each subsequent attempt
+ * @param verbose Whether to log extra info
  */
 export async function withExponentialBackoff<T>(
+  /**
+   * Function to call repeatedly
+   */
```

Comment:
echoing above: I can't imagine repeating yourself is required here :skull: but if it is, this format for one-liners is less noisy
```suggestion
  /** Function to call repeatedly */
```